### ISSUE TITLE
Unable to submit form despite validation passing.

### ISSUE THREAD

**Issue by turtletowerz** [fyne-io/fyne#3961]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When opening a form with a data-binded value, and specifically with regex validation (i have yet to test any other method) the form will occasionally open with a grayed out submit button even if the form input is validated. This behavior only happens when the form is opened, and after re-entering the value the validation fixes and the submit button becomes usable. 

### How to reproduce

1. Click "Change value" and change the value in the form to any number
2. Submit the form
3. Reopen the change value form and see that the submit button is grayed out even with validation passing (this is not guaranteed to happen on the first try, it seems pretty random)

### Screenshots

Example when this code is run, note the grayed out submit despite validation saying all form elements are passing.
![](https://cdn.discordapp.com/attachments/837686172494725160/1116086631913492730/image.png)

### Example code

```go
package main

import (
	"errors"
	"regexp"

	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/data/binding"
	"fyne.io/fyne/v2/widget"
)

func makeForm(a fyne.App, bind binding.String) {
	entry := widget.NewEntryWithData(bind)
	re := regexp.MustCompile(`\d+`)
	entry.Validator = func(s string) error {
		if !re.MatchString(s) {
			return errors.New("invalid input")
		}
		return nil
	}

	w := a.NewWindow("Value Changer")
	w.Resize(fyne.NewSize(200, 150))
	w.CenterOnScreen()

	form := widget.NewForm(
		widget.NewFormItem("Value", entry),
	)

	form.OnSubmit = func() {
		w.Close()
	}

	w.SetContent(container.NewVScroll(form))
	w.Show()
}

func main() {
	a := app.New()
	window := a.NewWindow("Bug Test")
	window.Resize(fyne.NewSize(200, 150))
	window.SetFixedSize(true)
	window.CenterOnScreen()
	window.SetMaster()

	var val int
	bind := binding.IntToString(binding.BindInt(&val))

	box := container.NewVBox(
		widget.NewButton("Change Value", func() {
			makeForm(a, bind)
		}),

		widget.NewLabel("Value"),
		widget.NewLabelWithData(bind),
	)

	window.SetContent(box)
	window.ShowAndRun()
}
```

### Fyne version

2.3.5

### Go compiler version

1.20.4

### Operating system and version

Arch Linux, Kernel 6.3.2

**Comment by Jacalz** [comment]
I can't seem to replicate this on `v2.3.5` or on the `develop` branch. Can you still replicate this?

**Comment by turtletowerz** [comment]
> I can't seem to replicate this on `v2.3.5` or on the `develop` branch. Can you still replicate this?

Yes, I can still reproduce this on 2.3.5. Takes me between 3 and 5 tries opening and closing the form window for it to occur. It doesn't seem to be consistent for smaller forms, but on any form with regex validation and 5+ elements it's much more likely to happen.

**Comment by andydotxyz** [comment]
What about on develop, or the v2.4.0-rc2 tag we are currently testing?

**Comment by turtletowerz** [comment]
Apologies for the delay,  I can still repro this with v2.4.0 and v2.4.1

![image](https://github.com/fyne-io/fyne/assets/29441515/d7a7c402-c63d-403d-8e6d-f8ccac74d1d6)

This is in my `go.mod`
```
require fyne.io/fyne/v2 v2.4.1
```

**Comment by tomsquest** [comment]
Hi all,

I have the same problem here, and reproducible code.
⏩ Given a binding and an entry validator, when the bound value is **initially ok**, then a validation error is reported.

![image](https://github.com/user-attachments/assets/e8fff3bc-c652-4acf-b4c1-c4d0be9cad78)

#### Versions
Fyne 2.5.1
Go 1.23.1


#### Reproducible error


```
package main

import (
	"fmt"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/data/binding"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("Bug Test")

	myStringBinding := binding.NewString()
	_ = myStringBinding.Set("ok")

	entry := widget.NewEntryWithData(myStringBinding)

	// BUG: when opening the window, the input is invalid,
	// even if the bound value is valid
	entry.Validator = func(text string) error {
		fmt.Println("Validator called with:", text)

		if text != "ok" {
			return fmt.Errorf("must be ok")
		}

		return nil
	}

	form := widget.NewForm()
	form.Append("Value", entry)

	form.OnSubmit = func() {
		myValue, e := myStringBinding.Get()

		if e != nil {
			fmt.Println("binding Error: ", e)
		}

		fmt.Println("myValue", myValue)
	}

	form.Refresh()

	w.SetContent(form)
	w.ShowAndRun()
}
```

**Comment by andydotxyz** [comment]
Hmm. This may relate to the binding operating on a different thread causing an unexpected order operations on setup. Let's see if we can get this sorted in a bug fix, but if I'm right it will go away with the 2.6 threading changes.

**Comment by tomsquest** [comment]
Pretty sure it's related to threading: I got different outputs on each run on a more complex code.

**Comment by turtletowerz** [comment]
I've tested with the 2.6.0 beta and this appears to be fixed, so I will close it. Thank you!
